src/plugins/intel_gna/src/transformations/insert_copy_layer.hpp

```diff
@@ -73,6 +73,40 @@ class InsertCopyBeforeConcatLayer : public ngraph::pass::MatcherPass {
     InsertCopyBeforeConcatLayer();
 };
 
+/**
+ * @brief Inserts Copy layer before Broadcast if it is connected to the network input layer
+ * while skipping Reshape/Trivial transpose/Squeeze/Unsqueeze (non-functional) layers:
+ *
+ *   [Parameter]              [Parameter]
+ *     |                          |
+ *     |               =>      [Copy]
+ *     |                          |
+ *   [Broadcast]               [Broadcast]
+ *
+ *
+ * With non-functional layers:
+ *
+ *   [Parameter]              [Parameter]
+ *     |                          |
+ * [Non functional     =>   [Non functional]
+ *     |                          |
+ *     |               =>      [Copy]
+ *     |                          |
+ *   [Broadcast]             [Broadcast]
+ *
+ * Note: Changes is required due the issue with legacy transformations.
+ * Broadcast is converted to Tile and later in some specific cases Tile is eliminated
```

Comment:
Don't we need to add Copy layers for these particular cases? Can we  detect them in the transformation?

Reply:
I would need here check how dimensions are converted from Broadcast to Tile and later check the same conditions which are in Checked done during Tile to TileIe conversion.

Reply:
I inserting copy layer for the situation when broadcast will be remove from network.